### ISSUE TITLE
C# Dispatch .bot endpoint service name doesn't match what's expected by the Statup.cs file

### ISSUE THREAD

**Issue by JonathanFingold** [Microsoft/BotBuilder-Samples#606]
## Sample information
1. Sample type: **samples**
2. Sample language: **dotnetcore**
3. Sample name: **14.nlp-with-dispatch**

## Describe the bug
C# Dispatch .bot endpoint service name doesn't match what's expected by the Statup.cs file, causing the app to crash on launch.

## To Reproduce
1. Go to the [README](https://github.com/Microsoft/BotBuilder-Samples/blob/master/samples/csharp_dotnetcore/14.nlp-with-dispatch/README.md) for the sample.
1. Follow the steps in the readme:
   1. [Install BotBuilder tools](https://github.com/Microsoft/BotBuilder-Samples/blob/master/samples/csharp_dotnetcore/14.nlp-with-dispatch/README.md#install-botbuilder-tools).
   1. [Manually configure required services](https://github.com/Microsoft/BotBuilder-Samples/blob/master/samples/csharp_dotnetcore/14.nlp-with-dispatch/README.md#manually-configure-required-services).
      - First run `luis init` and `qnamaker init` without providing an app ID or kb ID, respectively. (I'll open a separate issues about this.)
   1. Run the bot in [Visual Studio](https://github.com/Microsoft/BotBuilder-Samples/blob/master/samples/csharp_dotnetcore/14.nlp-with-dispatch/README.md#visual-studio).
1. The app throws an exception in Startup.cs, line 74:
```text
System.InvalidOperationException
  HResult=0x80131509
  Message=The .bot file does not contain an endpoint with name 'development'.
  Source=NLP_With_Dispatch_Bot
  StackTrace:
   at NLP_With_Dispatch_Bot.Startup.ConfigureServices(IServiceCollection services) in C:\Users\v-jofing\source\repos\BotBuilder-Samples\samples\csharp_dotnetcore\14.nlp-with-dispatch\Startup.cs:line 77
```

## Expected behavior
Following the instructions should not cause an exception.

## Additional context
Here are the offending lines. One of these two files should be updated so this exception is not thrown:

**BotConfiguration.bot** (lines 5-12):
```json
        {
            "type": "endpoint",
            "name": "nlp-with-dispatch",
            "id": "http://localhost:3978/api/messages",
            "appId": "",
            "appPassword": "",
            "endpoint": "http://localhost:3978/api/messages"
        },
```

**Startup.cs** (lines 69-75):
```csharp
            // Retrieve current endpoint.
            var environment = _isProduction ? "production" : "development";
            var service = botConfig.Services.Where(s => s.Type == "endpoint" && s.Name == environment).FirstOrDefault();
            if (!(service is EndpointService endpointService))
            {
                throw new InvalidOperationException($"The .bot file does not contain an endpoint with name '{environment}'.");
            }
```

[bug]

**Comment by daveta** [comment]
My recommendation is we remove the default .bot file --  it's a distraction.  For this sample, msbot should be used to generate the .bot configuration.

**Comment by stevengum** [comment]
Fixed via #613, @daveta, @JasonSowers and @vishwacsena are coming up with a solution for [this comment](https://github.com/Microsoft/BotBuilder-Samples/issues/606#issuecomment-427461470).
